lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_accessor :client
+        attr_reader :config
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          return client if client
+
+          options = {}
+
+          if opts.key?('region')
+            options[:region] = opts['region']
+          end
+          if opts.key?('profile')
+            options[:profile] = opts['profile']
+          end
+          if config.aws&.key?('credentials')
+            creds = File.expand_path(config.aws['credentials'])
+            if File.exist?(creds)
+              options[:credentials] = ::Aws::SharedCredentials.new(path: creds)
+            else
+              @logger.warn("Cannot load credentials file #{config.aws['credentials']}")
```

Comment:
I think if you specified an alternate credentials file and we don't find it we should error instead of warn. While it could theoretically fall back to a valid credentials file, the user (presumably) intended to use the credential file specified. Rather than behaving in a way the user didn't intend we should error and let the user either remove the aws config if they are not using it, or correct it if it's incorrect.